app/controllers/discovered_hosts_controller.rb

```diff
@@ -25,8 +26,19 @@ def index
 
   def show
     # filter graph time range
+    @general =    {:Highlights => ["size", "productname", "memorysize", "interface", "manufacturer", "architecture", "macaddress", "processorcount", "physicalprocessorcount", "discovery_boot"]}
+    @highlights = {:Storage => ["blockdevice"],
+                   :Hardware => ["hardw", "manufacturer", "memo", "process"],
+                   :Network => ["ipaddress", "interfaces", "dhcp", "fqdn", "hostname", "link", "mtu", "net"] + @host.facts_hash["interfaces"].split(","),
+                   :Software => ["bios", "os", "discovery"]}
+    @facts = @host.facts_hash
+    @facts.delete("memorysize_mb")
```

Comment:
can i make @facts a shallow copy of the facts hash and then the source will not be influenced?